pkg/testframework/console.go

```diff
@@ -0,0 +1,162 @@
+package testframework
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	// operatorapi "github.com/openshift/api/operator/v1"
+	operatorsv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	v1alpha1 "github.com/openshift/console-operator/pkg/apis/console/v1alpha1"
+)
+
+func isOperatorManaged(cr *v1alpha1.Console) bool {
+	return cr.Spec.ManagementState == operatorsv1alpha1.Managed
+}
+
+func isOperatorUnmanaged(cr *v1alpha1.Console) bool {
+	return cr.Spec.ManagementState == operatorsv1alpha1.Unmanaged
+}
+
+func isOperatorRemoved(cr *v1alpha1.Console) bool {
+	return cr.Spec.ManagementState == operatorsv1alpha1.Removed
+}
+
+type operatorStateReactionFn func(cr *v1alpha1.Console) bool
+
+func ensureConsoleIsInDesiredState(logger Logger, client *Clientset, state operatorsv1alpha1.ManagementState) error {
```

Comment:
So `ensureConsoleIsInDesiredState`leads me to think ALL console state, not just `ManagementState`.   I would probably expect the world `ensure` does an `UPDATE` as well to set the state if it is not already correct, but it looks like the `PATCH` is in the functions wrapping this call. 

Thinking about this.

Reply:
yeah, probably the wording is not the best.
What about `isConsoleInDesiredState` || `checkConsoleForDesiredState` ?

Reply:
or `isManagementStateManaged()`, `isManagementStateRemoved()`?